datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaBasedConnectorTaskMetrics.java

```diff
@@ -151,6 +201,11 @@ public void deregisterMetrics() {
       DYNAMIC_METRICS_MANAGER.unregisterMetric(_className, _key, TIME_SPENT_BETWEEN_POLLS_MS);
       DYNAMIC_METRICS_MANAGER.unregisterMetric(_className, _key, PER_EVENT_PROCESSING_TIME_NANOS);
     }
+
+    if (_kafkaConsumerMetricsRegistered) {
```

Comment:
I don't have a strong opinion on the matter but `unregisterMetric` no-ops if the metrics hasn't been registered. This may spare you having to maintain and check `_kafkaConsumerMetricsRegistered`.

Reply:
Removed the extra check and removed the boolean